app/StateToken.php

```diff
@@ -24,6 +27,73 @@ class StateToken extends Model
     protected $hidden = [
     ];
 
+    /**
+     * StateToken constructor. Always end up with a UUID
+     * @param array $attributes
+     */
+    public function __construct(array $attributes = [])
```

Comment:
I think if you set up the stateToken uuid to be constrained as unique - you'd be able to rely on $token->save() failing when you try to generate it and not need to do some of these extra checks.

Reply:
Fair.